vissl/engines/benchmark_suite_scheduler.py

```diff
@@ -0,0 +1,545 @@
+    and launch evaluations as checkpoints become available. The method takes a
+    config dictionary consisting of the training checkpoint directory, an array of
+    benchmarks, and information on how often to evaluate the trainings.
+    """
+
+    def __init__(
+        self,
+        training_checkpoint_dir: str,
+        benchmarks: List,
+        evaluate_final_phase=True,
+        evaluation_phase_freq: int = -1,
+        evaluation_iter_freq: int = -1,
+        max_training_iterations: int = None,
+        autoload_slurm_evaluator_checkpoint=False,
+        slurm_evaluator_checkpoint: str = None,
+        retry_evaluation_job_ids: List[int] = None,
+        auto_retry_evaluations=False,
+        max_retries=3,
+    ):
+        """
+        Args:
+
+        training_checkpoint_dir: (str). Checkpoint directory of the training.
+                                This should match the trainings CHECKPOINT.dir
+        benchmarks: (list[dict]) Benchmarks with the following structure:
+            "config_files": [
+                    {
+                        # Path to config file.
+                        "config=test/integration_test/quick_eval_in1k_linear.yaml"
+                        # Config overrides.
+                        "config.TRAIN.DATA_LIMIT=1000",
+                        ...
+                    },
+                    ...
+                ]
+        evaluate_final_phase: (bool, optional, default=True). Whether or not to evaluate the
+                                final phase of the training.
+        evaluation_phase_freq: (int, optional, default=-1) How often to evaluate phases.
+                                Training checkpoint phase freq must evenly
+                                divide evaluation_phase_freq.
+        evaluation_iter_freq: (int, optional, default=-1) How often to evaluate iterations.
+                               Training checkpoint iteration freq must evenly
+                               divide evaluation_iter_freq.
+        autoload_slurm_evaluator_checkpoint: (bool, optional, default=False) Whether or not to
+                                              autoload slurm_evaluator Checkpoint.
+                                              This is useful when slurm evaluator job
+                                              is preempted for example.
+        slurm_evaluator_checkpoint: (str, optional, default=None) String of
+                                    slurm_evaluator checkpoint directory.
+        retry_evaluation_job_ids: (List[int], optional, default=[]) List of job_ids to retry.
+        auto_retry_evaluations: (bool, optional, default=False) Whether or not to automatically
+                                retry all failed jobs.
+        max_retries: (int, optional, default=3). Maximum number of retries.
+        """
+        self.evaluation_jobs_finished = set()
+
+        # Required Arguments
```

Comment:
for all the required arguments, we must perform some kind of validation. In the `__init__` , you can call a `self.validate()` for instance that can take care of this.